AmplifyPlugins/API/AWSAPICategoryPlugin/Interceptor/RequestInterceptor/APIKeyURLRequestInterceptor.swift

```diff
@@ -21,7 +21,10 @@ struct APIKeyURLRequestInterceptor: URLRequestInterceptor {
         var modifiedRequest = request
         let apiKey = apiKeyProvider.getAPIKey()
         modifiedRequest.addValue(apiKey,
-                                 forHTTPHeaderField: URLRequestContants.Header.xApiKey)
+                                 forHTTPHeaderField: URLRequestConstants.Header.xApiKey)
+        modifiedRequest.setValue(AmplifyAWSServiceConfiguration.baseUserAgent(),
```

Comment:
Nit: for methods that take no value, do not throw, and return a value, prefer a computed property

Reply:
Per offline discussion, baseUserAgent static property was overriding the static function defined in the base class, but this was too confusing, so we are changing it a function w/ the override keyword.